### PR TITLE
INTYGFV-15115: Introduced abstract question class IntygBaseratPa in lisjp and ag7804 and for UntlatandeBaseratPa in luae_fs and luae_na.

### PR DESCRIPTION
**Pull request by mikeyOne**

### CODE REVIEW

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java [hunk 1]
```diff
@@ -40,23 +40,26 @@ public final class RespConstants {
     public static final String GRUNDFORMEDICINSKTUNDERLAG_TYP_DELSVAR_ID_1 = "1.1";
     public static final String GRUNDFORMEDICINSKTUNDERLAG_DATUM_DELSVAR_ID_1 = "1.2";
     public static final String GRUNDFORMEDICINSKTUNDERLAG_ANNANBESKRIVNING_DELSVAR_ID_1 = "1.3";
+    public static final String GRUNDFORMEDICINSKTUNDERLAG_ANNANBESKRIVNING_DELSVAR_TEXT = "DFR_1.3.RBK";
+    public static final String GRUNDFORMEDICINSKTUNDERLAG_BESKRIVNING_DELSVAR_JSON_ID_1 = "annatGrundForMUBeskrivning";
+    public static final String MOTIVERING_TILL_INTE_BASERAT_PA_UNDERLAG_ID_1 = "motiveringTillInteBaseratPaUndersokning";
+    public static final String GRUNDFORMEDICINSKTUNDERLAG_CATEGORY_TEXT = "KAT_1.RBK";
```
Comment (mww-sopra-steria) [inline fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java:46]: Prefix textIds with _ID

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java [hunk 2]
```diff
@@ -40,23 +40,26 @@ public final class RespConstants {
...
-    public static final String MOTIVERING_TILL_INTE_BASERAT_PA_UNDERLAG_ID_1 = "motiveringTillInteBaseratPaUndersokning";
-    public static final String GRUNDFORMEDICINSKTUNDERLAG_CATEGORY_TEXT = "KAT_1.RBK";
+    public static final String GRUNDFORMEDICINSKTUNDERLAG_ANNAT_SVAR_JSON_ID_1 = "annatGrundForMU";
+    public static final String GRUNDFORMU_UNDERSOKNING_LABEL_ID = "KV_FKMU_0001.UNDERSOKNING.RBK";
+    public static final String GRUNDFORMU_JOURNALUPPGIFTER_LABEL_ID = "KV_FKMU_0001.JOURNALUPPGIFTER.RBK";
+    public static final String GRUNDFORMU_ANHORIG_BESKRIVNING_LABEL_ID = "KV_FKMU_0001.ANHORIG.RBK";
+    public static final String GRUNDFORMU_TELEFONKONTAKT_LABEL_ID = "KV_FKMU_0001.TELEFONKONTAKT.RBK";
+    public static final String GRUNDFORMU_ANNAT_LABEL_ID = "KV_FKMU_0001.ANNAT.RBK";
+
     public static final String GRUNDFORMEDICINSKTUNDERLAG_SVAR_TEXT = "FRG_1.RBK";
```
Comment (mww-sopra-steria) [inline fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java:59]: Prefix textIds with _ID (I know it is not part of the PR, but maybe fix at the same time anyway :-) )

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java [hunk 3]
```diff
@@ -40,23 +40,26 @@ public final class RespConstants {
...
     public static final String GRUNDFORMEDICINSKTUNDERLAG_SVAR_BESKRIVNING = "FRG_1.HLP";
-    public static final String GRUNDFORMEDICINSKTUNDERLAG_ANNANBESKRIVNING_DELSVAR_TEXT = "DFR_1.3.RBK";
-    public static final String GRUNDFORMU_JOURNALUPPGIFTER_LABEL = "KV_FKMU_0001.JOURNALUPPGIFTER.RBK";
-    public static final String GRUNDFORMU_TELEFONKONTAKT_LABEL = "KV_FKMU_0001.TELEFONKONTAKT.RBK";
     public static final String GRUNDFORMU_ANHORIG_BESKRIVNING_LABEL = "KV_FKMU_0001.ANHORIG.RBK";
```
Comment (mww-sopra-steria) [inline fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java:61]: Prefix textIds with _ID (I know it is not part of the PR, but maybe fix at the same time anyway :-) )

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java [hunk 4]
```diff
@@ -40,23 +40,26 @@ public final class RespConstants {
...
-    public static final String GRUNDFORMEDICINSKTUNDERLAG_CATEGORY_TEXT = "KAT_1.RBK";
+    public static final String GRUNDFORMEDICINSKTUNDERLAG_ANNAT_SVAR_JSON_ID_1 = "annatGrundForMU";
+    public static final String GRUNDFORMU_UNDERSOKNING_LABEL_ID = "KV_FKMU_0001.UNDERSOKNING.RBK";
+    public static final String GRUNDFORMU_JOURNALUPPGIFTER_LABEL_ID = "KV_FKMU_0001.JOURNALUPPGIFTER.RBK";
+    public static final String GRUNDFORMU_ANHORIG_BESKRIVNING_LABEL_ID = "KV_FKMU_0001.ANHORIG.RBK";
+    public static final String GRUNDFORMU_TELEFONKONTAKT_LABEL_ID = "KV_FKMU_0001.TELEFONKONTAKT.RBK";
+    public static final String GRUNDFORMU_ANNAT_LABEL_ID = "KV_FKMU_0001.ANNAT.RBK";
+
     public static final String GRUNDFORMEDICINSKTUNDERLAG_SVAR_TEXT = "FRG_1.RBK";
     public static final String GRUNDFORMEDICINSKTUNDERLAG_SVAR_BESKRIVNING = "FRG_1.HLP";
```
Comment (mww-sopra-steria) [inline fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java:60]: Prefix textIds with _ID and call it DESCRIPTION i.e.  GRUNDFORMEDICINSKTUNDERLAG_SVAR_DESCRIPTION_ID (I know it is not part of the PR, but maybe fix at the same time anyway :-) )

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/certificate/AbstractQuestionIntygetBaseratPa.java [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+import se.inera.intyg.common.support.facade.model.value.CertificateDataValueDate;
+import se.inera.intyg.common.support.facade.model.value.CertificateDataValueDateList;
+import se.inera.intyg.common.support.model.InternalDate;
+
+public abstract class AbstractQuestionIntygetBaseratPa {
+
+    private static final short NUMBER_OF_DAYS_IN_FUTURE = 0;
+
+    protected static CertificateDataElement toCertificate(InternalDate undersokningPatient, InternalDate telefonkontaktPatient,
+        InternalDate journaluppgifter, InternalDate annat, String questionId, String parent, int index,
```
Comment (mww-sopra-steria) [inline fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/certificate/AbstractQuestionIntygetBaseratPa.java:58]: parent -> parentId

File: fk/lisjp/src/main/java/se/inera/intyg/common/lisjp/v1/model/converter/RespConstants.java [hunk 6]
```diff
@@ -21,6 +21,18 @@
 
 public class RespConstants {
 
+    public static final String GRUNDFORMEDICINSKTUNDERLAG_SVAR_ID_1 = "1";
+    public static final String GRUNDFORMU_CATEGORY_ID = "grundformu";
+    public static final String GRUNDFORMEDICINSKTUNDERLAG_SVAR_TEXT = "FRG_1.RBK";
```
Comment (mww-sopra-steria) [inline fk/lisjp/src/main/java/se/inera/intyg/common/lisjp/v1/model/converter/RespConstants.java:26]: Prefix textIds with _ID.

File: fk/lisjp/src/main/java/se/inera/intyg/common/lisjp/v1/model/converter/RespConstants.java [hunk 7]
```diff
@@ -21,6 +21,18 @@
...
+    public static final String GRUNDFORMEDICINSKTUNDERLAG_SVAR_BESKRIVNING = "FRG_1.HLP";
```
Comment (mww-sopra-steria) [inline fk/lisjp/src/main/java/se/inera/intyg/common/lisjp/v1/model/converter/RespConstants.java:27]: _BESKRIVNING to _DESCRIPTION_ID.

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Awesome! A few comments on naming to follow naming-convention. I noticed a few things to rename that wasn't introduced in this PR, but as they are in the same file maybe it could be dealt with here. Hope you don't disapprove :-)

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java [hunk 8]
```diff
@@ -40,23 +40,26 @@ public final class RespConstants {
     public static final String GRUNDFORMEDICINSKTUNDERLAG_TYP_DELSVAR_ID_1 = "1.1";
     public static final String GRUNDFORMEDICINSKTUNDERLAG_DATUM_DELSVAR_ID_1 = "1.2";
     public static final String GRUNDFORMEDICINSKTUNDERLAG_ANNANBESKRIVNING_DELSVAR_ID_1 = "1.3";
+    public static final String GRUNDFORMEDICINSKTUNDERLAG_ANNANBESKRIVNING_DELSVAR_TEXT = "DFR_1.3.RBK";
```
[the remark this answers is not in the document]
Comment (mikeyOne) [reply fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java:43]: Done

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java (hunk 1 above)
Comment (mikeyOne) [reply fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java:46]: DOne

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java (hunk 2 above)
Comment (mikeyOne) [reply fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java:59]: Mot used -> removed

Actually had this block of constants that weren't used -> removed
public static final String GRUNDFORMEDICINSKTUNDERLAG_SVAR_TEXT = "FRG_1.RBK";
    public static final String GRUNDFORMEDICINSKTUNDERLAG_SVAR_BESKRIVNING = "FRG_1.HLP";
    public static final String GRUNDFORMU_ANHORIG_BESKRIVNING_LABEL = "KV_FKMU_0001.ANHORIG.RBK";
    public static final String GRUNDFORMEDICINSKTUNDERLAG_SVAR_TEXT_ID = "FRG_1.RBK";

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java (hunk 4 above)
Comment (mikeyOne) [reply fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java:60]: Removed in block

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java (hunk 3 above)
Comment (mikeyOne) [reply fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/RespConstants.java:61]: Removed in block

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/certificate/AbstractQuestionIntygetBaseratPa.java (hunk 5 above)
Comment (mikeyOne) [reply fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/certificate/AbstractQuestionIntygetBaseratPa.java:58]: Done

File: fk/lisjp/src/main/java/se/inera/intyg/common/lisjp/v1/model/converter/RespConstants.java (hunk 6 above)
Comment (mikeyOne) [reply fk/lisjp/src/main/java/se/inera/intyg/common/lisjp/v1/model/converter/RespConstants.java:26]: Done

**Review by mww-sopra-steria — APPROVED** [review]
Approved!

**Pull request merged**